gstd/src/common/primitives.rs

```diff
@@ -236,6 +266,39 @@ impl From<H256> for MessageId {
     }
 }
 
+impl core::fmt::Display for MessageId {
+    fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
+        let len = self.0.len();
+        let median = (len + 1) / 2;
+
+        let mut e1 = median;
+        let mut s2 = median;
+
+        if let Some(precision) = f.precision() {
+            if precision < median {
+                e1 = precision;
+                s2 = len - precision;
+            }
+        }
+
+        let p1 = hex::encode(&self.0[..e1]);
+        let p2 = hex::encode(&self.0[s2..]);
+        let sep = e1.ne(&s2).then_some("..").unwrap_or_default();
+
+        if f.alternate() {
+            write!(f, "MessageId(0x{p1}{sep}{p2})")
+        } else {
+            write!(f, "0x{p1}{sep}{p2}")
```

Comment:
I think you can move this functionality out into a separate function passing [u8;32] + &str for alternate + fmt and use for both ActorId and MessageId